src/containers/Header/NotificationsTray/index.jsx

```diff
@@ -0,0 +1,74 @@
+import styles from './style.postcss';
+
+import React, { PureComponent } from 'react';
+import Cursor from 'immutable-cursor';
+import Config from 'domain/Config';
+import moment from 'domain/moment';
+import connect from 'domain/connect';
+import Icon from 'components/Icon';
+import Callout from 'components/Callout';
+
+class NotificationsTray extends PureComponent {
+  constructor() {
+    super();
+    this.state = { open: false };
+    this._getRef = this._getRef.bind(this);
+  }
+
+  _getRef(node) {
```

Comment:
Just a curiosity: the `ref` prop is not a normal React prop, and it shouldn't be causing the lifecycle to be called again. I'm not 100% sure, but we could double-check in case we can avoid unnecessary “code unfolding”.